### ISSUE TITLE
Dep `rollup-plugin-inject-process-env` in v0.1.16+ makes SDK unusable in standard Vue / webpack setup

### ISSUE THREAD

**Issue by santi** [millicast/millicast-sdk#122]
**Version affected**
v0.1.16

**Describe the bug**
The newly introduced plugin `rollup-plugin-inject-process-env` (used to inject env variables during build) is incompatible with webpack production builds using common config setups. 

This is due to `rollup-plugin-inject-process-env` injecting code at the top of output modules, with the following line: 
`process.env = Object.assign({}, process.env);`

This line is further processed by webpack, where it by default replaces references to `process.env` with an object containing the actual env variables. The line of code from rollup is thus converted into: `{NODE_ENV:"production",MORE_ENV_VARIABLES:"example"} = Object.assign({},{NODE_ENV:"production",MORE_ENV_VARIABLES:"example"})`

Which is not valid JS (you are basically trying to destructure the result from `Object.assign` into an object with values). 

Note: This is NOT a problem during development config with webpack, only when building for production / with minification enabled.

**Proposed fix**
Use another library or method to use env variables in code. This library is also reported to have [problems with standard setups of Create React app](https://github.com/badcafe/rollup-plugin-inject-process-env/issues/13): 

**To Reproduce**
Example repo: 
[Rollup env transpilation error](https://github.com/santi/rollup-env-transpilation-error)

1. Create a basic Vue 3 project with Vue CLI: `npx @vue/cli create example-project`
2. Install Millicast SDK: `cd example-project && yarn add @millicast/sdk`
3. Add an import from Millicast SDK and use it somehow: `import {View} from "@millicast/sdk"; console.log(View);`
4. Try to build project: `yarn build`
5. Expected error: 
```
js/chunk-vendors.5cc891d8.js from Terser
Invalid assignment [./node_modules/@millicast/sdk/dist/millicast.umd.js:1,647][js/chunk-vendors.5cc891d8.js:13296,572]
```

**Expected behavior**
Minified build including the Millicast SDK library

**Screenshots**

**Environment (please complete the following information):**
 - OS: macOS 12.0.1
 - Browser: Node
 - Version 16/17 (unrelated)

**Additional context**

**Comment by R-Delfino95** [comment]
Thanks for report it. Yes, we should change how we inject the environment variables, we are searching for an alternative.

For now you can use the [v0.1.15](https://github.com/millicast/millicast-sdk/releases/tag/v0.1.15) that has the same features as the new one.

**Comment by santi** [comment]
Great 👍 

If I understand this correctly, we do not want any environment variables injected when publishing this library, only when using it as a library in a bigger application, built with a common tool like webpack, rollup, esbuild etc...

However, a client needs to be able to consume this library as a pure script when loading it from a CDN, se we still need to provide valid JS when executing it from a normal `<script>` tag. 

If my assumptions above are correct, then we should just keep the `process.env.X` entries in our .cjs and .esm outputs, while replacing them with `undefined` in the `.umd` output (which is the output used by your CDN). Bundlers will then be able to replace instances of `process.env` while bundling the consuming app, and using the UMD build from the CDN directly will fallback to the default values (by using `||`).

I suggest using [@rollup/plugin-replace](https://www.npmjs.com/package/@rollup/plugin-replace) on the UMD build to replace the `process.env` references with `undefined`.

**Comment by santi** [comment]
@R-Delfino95 I made a PR which implements my proposed solution above, feel free to check it out: https://github.com/millicast/millicast-sdk/pull/123

**Comment by cantuccis** [comment]
Hi @santi, I reviewed your fix, it solves the issue.
We have some ci/cd processes that depend on those variables to configure different builds of the SDK. 
Some more changes from what you did will be required. I'll take care of that.

Thanks for reporting the issue and for the fix!

**Comment by R-Delfino95** [comment]
> @R-Delfino95 I made a PR which implements my proposed solution above, feel free to check it out: #123

Hi @santi, we just released the [v0.1.17](https://github.com/millicast/millicast-sdk/releases/tag/v0.1.17) that removes the `rollup-plugin-inject-process-env` and uses a json configuration instead of .env. Now you should be able to use the Webpack minification without trouble.
I'm also going to close your PR because we don't use environments so it's no necessary to replace it.

Thanks again for report this issue.

**Comment by santi** [comment]
@R-Delfino95 Great! Thanks a lot for the fix 🙌
